flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/nonha/embedded/EmbeddedJobResultStoreTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability.nonha.embedded;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.clusterframework.ApplicationStatus;
+import org.apache.flink.runtime.highavailability.JobResultEntry;
+import org.apache.flink.runtime.jobmaster.JobResult;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.NoSuchElementException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/** Tests for the {@link EmbeddedJobResultStore}. */
+public class EmbeddedJobResultStoreTest extends TestLogger {
+
+    private EmbeddedJobResultStore embeddedJobResultStore;
+
+    private JobResult jobResult;
+
+    @Before
+    public void setupTest() {
+        embeddedJobResultStore = new EmbeddedJobResultStore();
+        jobResult =
+                new JobResult.Builder()
+                        .applicationStatus(ApplicationStatus.UNKNOWN)
```

Comment:
We shouldn't test stuff that should not occur in production. `ApplicationStatus.UNKNOWN` would be an invalid state because it means that the was not globally-terminated. That makes me thinking whether we should add this as a Precondition to the `EmbeddedJobResultStore`. WDYT? 🤔

Reply:
Aha, great point. I think I just used a pattern of creating job results that I found elsewhere in testing. Do you mean that if I user tries to create a dirty result from a jobresult that isnt in `SUCCEEDED | CANCELLED | FAILED` it should throw an exception? Would it make sense to embed this assumption into the interface as well, or do you think an implementation might have reason not to do that?

Reply:
Even though I like it making it as a general rule, it would make the interface more verbose if we implemented in `JobResultStore`. I'd be ok with mentioning the contract in the JavaDoc and adding it in the implementing classes.